Micro Trend One Vision Event Collector (#26650)
demisto/content: Packs/TrendMicroVisionOne/Integrations/TrendMicroVisionOneEventCollector/README.md

> Palo Alto Networks Trend Micro Vision One Event Collector integration for XSIAM, collects the Workbench, Observed Attack Techniques, Search Detections and Aduit logs.

Reviewer: ```suggestion
Palo Alto Networks Trend Micro Vision One Event Collector integration for Cortex XSIAM collects the Workbench, Observed Attack Techniques, Search Detections and Audit logs.
```

> Trend Micro Vision One is a purpose-built threat defense platform that provides added value and new benefits beyond XDR solutions, allowing you to see more and respond faster. Providing deep and broad extended detection and response (XDR) capabilities that collect and automatically correlate data across multiple security layers—email, endpoints, servers, cloud workloads, and networks—Trend Micro Vision One prevents the majority of attacks with automated protection.

Reviewer: ```suggestion
Trend Micro Vision One is a purpose-built threat defense platform that provides added value and new benefits beyond Cortex XDR solutions, allowing you to see more and respond faster. Providing deep and broad extended detection and response (Cortex XDR) capabilities that collect and automatically correlate data across multiple security layers—email, endpoints, servers, cloud workloads, and networks—Trend Micro Vision One prevents the majority of attacks with automated protection.
```

1. Navigate to **Settings** > **Configurations** > **Data Collection** > **Automation & Feed Integrations**.
2. Search for Trend Micro Vision One Event Collector.
3. Click **Add instance** to create and configure a new integration instance.

| **Parameter** | **Description** | **Required** |
| --- | --- | --- |
| Your server URL | The api endpoint to the trend micro vision one instance, see domains list: https://automation.trendmicro.com/xdr/Guides/First-Steps-Toward-Using-the-APIs | True |
| Trend Micro Vision One API Key | The Trend Micro Vision One API Key. Refer to the help section or to the information below on how to retrieve the API key. | False |
| The maximum number of events per fetch | The maximum number of events to fetch every time fetch is executed for a single log-type \(Workbench, Observed Attack Techniques, Search Detections and Audit logs\). | False |
| Observed attack techniques and Search detections logs date-time range (hours) | Defines the date-range (hours) in each api call that retrieves observed attack techniques and search detections logs. Used mainly to prevent timeouts by looping over large amount of logs.  Default is 24 hours. | False |

> | First fetch timestamp (&lt;number&gt; &lt;time unit&gt;, e.g., 12 hours, 7 days) |  | False |

Reviewer 2: lets add descriptions to these

Author: its auto generated by `demisto-sdk generate-docs`, what should i add here?

Reviewer 2: see what other integrations use, theyre pretty standard

### Api Limitations
* You cannot retrieve audit logs that are older than 180 days, hence if setting a first fetch that is more than 180 days, for audit-logs it will be maximum 180 days.
* For api rate limits please refer [here](https://automation.trendmicro.com/xdr/Guides/API-Request-Limits)

## Commands

You can execute these commands from the Cortex XSIAM CLI, as part of an automation, or in a playbook.
After you successfully execute a command, a DBot message appears in the War Room with the command details.

### trend-micro-vision-one-get-events

***
Returns a list of logs.

#### Base Command

`trend-micro-vision-one-get-events`

#### Input

| **Argument Name**  | **Description** | **Required** |
|--------------------| --- | --- |
| limit              | The maximum number of logs to retrieve. Default is 50. | Optional | 
| from_time          | From which time to retrieve the log(s) (&lt;number&gt; &lt;time unit&gt;, for example 12 hours, 1 day, 3 months). Default is 3 days. | Optional | 
| to_time            | To which time to retrieve the log(s), a date in ISO8601 format, will default to current time if not provided. | Optional | 

> | should_push_events | Whether to push the fetched events to XSIAM or not. Possible values are: false, true. Default is false. | Optional | 

Reviewer: ```suggestion
| should_push_events | Whether to push the fetched events to Cortex XSIAM or not. Possible values are: false, true. Default is false. | Optional | 
```
